extensions/identity-hub/src/test/java/org/eclipse/dataspaceconnector/identityhub/api/featuredetection/IdentityHubControllerTest.java

```diff
@@ -0,0 +1,142 @@
+import org.bouncycastle.util.encoders.Base64;
+import org.eclipse.dataspaceconnector.identityhub.dtos.Descriptor;
+import org.eclipse.dataspaceconnector.identityhub.dtos.MessageRequestObject;
+import org.eclipse.dataspaceconnector.identityhub.dtos.RequestObject;
+import org.eclipse.dataspaceconnector.identityhub.dtos.VerifiableCredential;
+import org.eclipse.dataspaceconnector.junit.launcher.EdcExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.identityhub.dtos.WebNodeInterfaces.COLLECTIONS_QUERY;
+import static org.eclipse.dataspaceconnector.identityhub.dtos.WebNodeInterfaces.COLLECTIONS_WRITE;
+import static org.eclipse.dataspaceconnector.identityhub.dtos.WebNodeInterfaces.FEATURE_DETECTION_READ;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+@ExtendWith(EdcExtension.class)
+public class IdentityHubControllerTest {
+
+    private static final String API_URL = "http://localhost:8181/api";
+    private static final Faker FAKER = new Faker();
+    private static final String VERIFIABLE_CREDENTIAL_ID = FAKER.internet().uuid();
+    private static final String NONCE = FAKER.lorem().characters(32);
+    private static final String TARGET = FAKER.internet().url();
+    private static final String REQUEST_ID = FAKER.internet().uuid();
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Test
+    void pushAndQueryVerifiableCredentials() throws IOException {
+        VerifiableCredential credential = VerifiableCredential.Builder.newInstance().id(VERIFIABLE_CREDENTIAL_ID).build();
+
+        pushVerifiableCredential(credential);
+        List<VerifiableCredential> verifiableCredentials = queryVerifiableCredentials();
+
+        assertThat(verifiableCredentials).usingRecursiveFieldByFieldElementComparator().containsExactly(credential);
+    }
+
+    private void pushVerifiableCredential(VerifiableCredential credential) throws IOException {
+        byte[] data = Base64.encode(OBJECT_MAPPER.writeValueAsString(credential).getBytes(StandardCharsets.UTF_8));
+        baseRequest()
+            .body(createRequestObject(COLLECTIONS_WRITE, data))
+            .post()
+        .then()
+            .statusCode(200)
+            .body("requestId", equalTo(REQUEST_ID))
+            .body("replies", hasSize(1))
+            .body("replies[0].status.code", equalTo(200))
+            .body("replies[0].status.detail", equalTo("The message was successfully processed"));
+    }
+
+    private List<VerifiableCredential> queryVerifiableCredentials() {
```

Comment:
question: why use REST Assured if we have a generated client? Wouldn't the latter make tests easier to maintain?

Reply:
In story https://github.com/agera-edc/MinimumViableDataspaceFork/issues/5 we will create the client and adapt the tests to use the CredentialsVerifier for reading credentials